### PR TITLE
add new logging defaults to default config

### PR DESCRIPTION
**Pull request by garbados**
## Proposed Release Notes

- Enable log forwarding by default in the example config.

## Links

- Closes https://github.com/newrelic/node-newrelic/issues/1150

## Details

See also: https://github.com/newrelic/node-newrelic/pull/1158

This PR is in draft because I can't find `local_decorating` in the agent spec, and thus am not sure how to write a descriptive comment about it. What is an appropriate descriptor for its specific function?

### CODE REVIEW

File: newrelic.js [hunk 1]
```diff
@@ -44,6 +44,18 @@ exports.config = {
    * attributes include/exclude lists.
    */
   allow_all_headers: true,
+  application_logging: {
+    /**
+     * When true, logs generated by your application will be forwarded to New Relic One.
```
Comment (bizob2828) [inline newrelic.js:49]: this is not techically true.  you need to have both `application_logging.enabled` and `application_logging.forwarding.enabled` set to true.  The ticket #1150 describes what to add to this default, this is missing the forwarding bit and included the application_logging.enabled flag even though that defaults to true in agent

File: newrelic.js [hunk 2]
```diff
@@ -44,6 +44,18 @@ exports.config = {
...
+     */
+    enabled: true,
+    local_decorating: {
+      /**
+       * TODO
```
Comment (bizob2828) [inline newrelic.js:54]: The spec references it here https://source.datanerd.us/agents/agent-specs/pull/570/files#diff-22ea975b3218ea7ba317770796151db7a25e9e8a59ef0799d8223ee8506802c9R122

File: newrelic.js (hunk 1 above)
Comment (garbados) [reply newrelic.js]: Updated!

File: newrelic.js (hunk 2 above)
Comment (garbados) [reply newrelic.js]: 👍

File: newrelic.js [hunk 3]
```diff
@@ -44,6 +44,38 @@ exports.config = {
...
+    forwarding: {
+      /**
+       * Toggles whether the agent gathers log records for sending to New Relic.
+       */
+      enabled: false,
+      /**
+       * Number of log records to send per minute to New Relic.
+       * Controls the overall memory consumption when using log forwarding.
+       */
+      max_samples_stored: 10000
```
Comment (bizob2828) [inline newrelic.js:62]: we don't intend for users to change this and if we do they can set it themselves.  I recommend re-reading the ticket

File: newrelic.js [hunk 4]
```diff
@@ -44,6 +44,38 @@ exports.config = {
...
+    },
+    metrics: {
+      /**
+       * Toggles whether the agent gathers the the user facing `Logging/lines`
+       * and `Logging/lines/{SEVERITY}` *Logging Metrics* used in the Logs chart
+       * on the APM Summary page.
+       */
+      enabled: true
```
Comment (bizob2828) [inline newrelic.js:70]: this is currently defaulted to true there's no reason to set this in here

File: newrelic.js [hunk 5]
```diff
@@ -44,6 +44,38 @@ exports.config = {
    * attributes include/exclude lists.
    */
   allow_all_headers: true,
+  application_logging: {
+    /**
+     * If this is disabled, all sub-features are disabled.
+     * If it is enabled, the individual sub-feature configurations take effect.
+     */
+    enabled: true,
```
Comment (bizob2828) [inline newrelic.js:52]: this is currently defaulting to true in code there's no reason to set this in here

File: newrelic.js (hunk 4 above)
Comment (garbados) [reply newrelic.js:52]: The ticket specifically identifies `application_logging.forwarding.enabled` to be set to true and described. Have I read this wrong?

**Review by jordigh — APPROVED** [review]
Matches the requirements, short and sweet.

**Review by bizob2828 — APPROVED** [review]

**Pull request merged**